### PR TITLE
Doc what promises resolve to. Consistent params. Fix English.

### PR DESCRIPTION
**Pull request by dandv**
* document what findAnd* promises resolve to
* fix run-on sentences
* replace "on MongoDB" with "in/from a collection"
* be specific re. update vs. replace
* consistent @param descriptions

### CODE REVIEW

File: lib/collection.js [hunk 1]
```diff
@@ -687,6 +687,8 @@ Collection.prototype.insert = deprecate(function(docs, options, callback) {
  * @property {Number} upsertedCount The number of documents upserted.
  * @property {Object} upsertedId The upserted id.
  * @property {ObjectId} upsertedId._id The upserted _id returned from the server.
+ * @property {Object} message TBD
+ * @propert {Array} ops TBD
```
Comment (dandv) [inline lib/collection.js]: Please document these after merging. See https://jira.mongodb.org/browse/NODE-1611

File: lib/collection.js [hunk 2]
```diff
@@ -746,7 +748,7 @@ Collection.prototype.updateOne = function(filter, update, options, callback) {
  * @param {boolean} [options.bypassDocumentValidation=false] Allow driver to bypass schema validation in MongoDB 3.2 or higher.
  * @param {ClientSession} [options.session] optional session to use for this operation
  * @param {Collection~updateWriteOpCallback} [callback] The command result callback
- * @return {Promise} returns Promise if no callback passed
+ * @return {Promise} returns Promise resolving to {Collection~updateWriteOpResultObject} if no callback passed
```
Comment (daprahamian) [inline lib/collection.js]: @dandv Could we instead describe the type as `Promise<Collection~updatewriteOpResultObject>`?

File: lib/collection.js (hunk 1 above)
Comment (daprahamian) [reply lib/collection.js]: typo on `@propert`

**Comment by dandv** [comment]
Let me know if there's anything I need to fix before this can be merged!

**Comment by daprahamian** [comment]
Hi @dandv ! Sorry for the long wait. Can you make one more change and remove the "TBD" from the new properties? Just leave it blank, and then we will fill it in.

**Comment by daprahamian** [comment]
thank you for the contribution @dandv !

**Pull request merged**